src/LuxCore.jl

```diff
@@ -106,11 +114,23 @@ Return the input size of the layer.
 function inputsize end
 
 """
-    outputsize(layer)
+    outputsize(layer, x, rng)
 
-Return the output size of the layer.
+
+Return the output size of the layer. If the output size can be statically determined
+(see [`has_static_outputsize`](@ref)), one can also use `outputsize(layer)` directly.
 """
-function outputsize end
+outputsize(layer, x, rng) = outputsize(has_static_outputsize(layer), x, rng)
+
+function outputsize(::Val{true}, x, rng)
+    outputsize(layer)
+end
+
+function outputsize(::Val{false}, x, rng)
+    ps, st = Lux.setup(rng, layer)
+    y = first(layer(x, ps, st))
+    size(y)
```

Comment:
this is not `size(y)` since `y` might not be a array. Here my recommendation would be to define `__size` and for some primitive types define it for eg:

```julia
__size(x::AbstractArray)
__size(x::Tuple)
__size(x::NamedTuple)
__size(x) = use fmap here
```

Also you need to remember to drop the last dim which is the batch dim

Reply:
I think that
```julia
__size(x::AbstractArray{T, N}) where {T} = isbitstype(T) ? size(x)[1:(N - 1)] : __size.(x)
```
needs to drop the last dim only if batch input is used.

how do I know that the batch input was used? 😅

Reply:
We can mention in the docs that `batch` is implied for `ndims(x) > 1`. If that doesn't work for the user, then they can always implement a custom `outputsize`